xds/src/main/java/com/linecorp/centraldogma/xds/application/v1/XdsApplicationService.java

```diff
@@ -0,0 +1,117 @@
+                                  StreamObserver<Application> responseObserver) {
+        final String applicationName = request.getApplication().getName();
+        final String name = removePrefix("applications/", applicationName);
+        if (projectManager.get(XDS_CENTRAL_DOGMA_PROJECT).repos().exists(name)) {
+            throwAlreadyExists(name);
+        }
+        createRepository(commandExecutor, mds, currentAuthor(), XDS_CENTRAL_DOGMA_PROJECT, name)
+                .handle((unused, cause) -> {
+                    if (cause != null) {
+                        if (cause instanceof RepositoryExistsException) {
+                            throwAlreadyExists(name);
+                        }
+                        responseObserver.onError(
+                                Status.INTERNAL.withCause(cause).asRuntimeException());
+                        return null;
+                    }
+                    responseObserver.onNext(Application.newBuilder().setName(applicationName).build());
+                    responseObserver.onCompleted();
+                    return null;
+                });
+    }
+
+    private static String removePrefix(String prefix, String name) {
+        if (!name.startsWith(prefix)) {
+            throw new StatusRuntimeException(
+                    Status.INVALID_ARGUMENT.withDescription(name + " does not start with prefix: " + prefix));
+        }
+        return name.substring(prefix.length());
+    }
+
+    private static void throwAlreadyExists(String applicationName) {
+        throw Status.ALREADY_EXISTS.withDescription("Application already exists: " + applicationName)
+                                   .asRuntimeException();
+    }
+
+    @Override
+    public void deleteApplication(DeleteApplicationRequest request, StreamObserver<Empty> responseObserver) {
+        final String applicationName = request.getName();
+        final String name = removePrefix("applications/", applicationName);
+        if (!projectManager.get(XDS_CENTRAL_DOGMA_PROJECT).repos().exists(name)) {
+            throw Status.NOT_FOUND.withDescription("Application does not exist: " + applicationName)
+                                  .asRuntimeException();
+        }
+        if (Project.isReservedRepoName(name)) {
+            throw Status.PERMISSION_DENIED.withDescription("Now allowed to delete " + applicationName)
+                                          .asRuntimeException();
+        }
+
+        // TODO(minwoox): Check the permission.
+        removeRepository(commandExecutor, mds, currentAuthor(), XDS_CENTRAL_DOGMA_PROJECT, name)
+                .handle((unused, cause1) -> {
+                    if (cause1 != null) {
+                        responseObserver.onError(
+                                Status.INTERNAL.withDescription("Failed to delete " + applicationName)
+                                               .withCause(cause1).asRuntimeException());
```

Comment:
Add `return` in the end of this block?
```suggestion
                                               .withCause(cause1).asRuntimeException());
                             return;
```

Reply:
Oops. 😓